packages/frontend/src/connectors/frame/index.ts

```diff
@@ -60,57 +84,126 @@ export class FrameConnector extends Connector<JsonRpcProvider, unknown> {
     }
 
     public async getChainId() {
         const provider = await this.getProvider();
-        const network = await provider.getNetwork();
-        const chainId = normalizeChainId(network.chainId);
-        return chainId;
+        if (!provider.chainId) throw new Error("no chain id in provider");
+        return normalizeChainId(provider.chainId);
     }
 
-    public async getProvider(config?: { chainId?: number }) {
+    public async getProvider(): Promise<Provider> {
         if (!this.provider) {
-            this.provider = new JsonRpcProvider(
-                "http://127.0.0.1:1248",
-                config?.chainId
-            );
+            this.provider = getProvider("frame");
             this.provider.on("accountsChanged", this.onAccountsChanged);
             this.provider.on("chainChanged", this.onChainChanged);
             this.provider.on("disconnect", this.onDisconnect);
         }
         return this.provider;
     }
 
-    async getSigner() {
-        const provider = await this.getProvider();
-        return provider.getSigner();
+    async getWalletClient({
+        chainId,
+    }: { chainId?: number } = {}): Promise<WalletClient> {
+        const [provider, account] = await Promise.all([
+            this.getProvider(),
+            this.getAccount(),
+        ]);
+        const chain = this.chains.find((x) => x.id === chainId);
+        if (!provider) throw new Error("provider is required.");
+        return createWalletClient({
+            account,
+            chain,
+            transport: custom(provider),
+        });
     }
 
     async isAuthorized() {
-        return true;
+        try {
+            return !!((await this.getProvider()) && (await this.getAccount()));
+        } catch {
+            return false;
+        }
     }
 
     async switchChain(chainId: number): Promise<Chain> {
-        await this.getProvider({ chainId });
-        const chain = this.chains.find((x) => x.id === chainId);
-        if (!chain)
-            throw new Error(`can't find switched chain with id ${chainId}`);
-        this.onChainChanged(chainId);
-        return chain;
+        const provider = await this.getProvider();
+        if (!provider) throw new ConnectorNotFoundError();
+        const targetChain = this.chains.find((chain) => chain.id === chainId);
+        if (!targetChain)
+            throw new Error(`chain with id ${chainId} is not supported`);
+        const hexChainId = numberToHex(chainId);
+
+        try {
+            await provider.request({
+                method: "wallet_switchEthereumChain",
+                params: [{ chainId: hexChainId }],
+            });
+            await new Promise<void>((resolve) =>
+                this.on("change", ({ chain }) => {
+                    if (chain?.id === chainId) resolve();
+                })
+            );
+            return targetChain;
+        } catch (error) {
+            if ((error as ProviderRpcError).code === 4902) {
```

Comment:
Should we add a dedicated function like `isUserRejectedRequestError` for this error too? So it's clear what type of error the code `4902` refers to.